GoogleForCreators/web-stories-wp
# Code Quality: Clean up Checkpoint Provider

## Context

When the prepublish checklist was initially done, there was a dialog that showed up asking if users were sure they were ready to publish. This dialog would show up based on if there were high priority issues present in their story or if they had yet to unlock the various checkpoints of it. 

Now we have a new modal that is seen when "publish" is clicked from the editor no matter what so this little dialog is unnecessary anymore. Most of the logic in the provider that was controlling when to have that dialog show up is actually still important to controlling the automatic opening of the checklist conditionally as well as the little icon and tooltip text for the publish buttons. So this is largely just renaming and doing some minor clean up. 

When I was first implementing the new modal I thought more of the checkpoint provider would get destroyed. 

## Summary

Renamed everything related to `reviewDialog` and made it more so about what was happening up stream that is influencing the flow - so `reviewChecklist`  or `hasHighPriorityIssues` - things that are easier to also have elsewhere should we need them without their naming losing meaning. 


## Relevant Technical Choices

Aside from some renaming, the only thing that's changing here is the `private` publish flow. This was the only place that the review checklist dialog would show up. I checked on how posts work with the private flow,  my update here has Story private publish working that same way. With the review checklist dialog you'd 1) select 'private' from the dropdown for visibility 2) get the default WP alert asking if you're sure, assuming you say yes 3) the review checklist dialog. Now you just get the first two steps. 

I did this because if you're in the story details modal and you hit 'private' that's already 2 things popped up in your browser - the story details modal and the new alert asking if you're sure. It just feels excessive. I talked a while ago with Amy about it and this is just killing it officially. 

If this isn't ok for some reason I don't know, we can put it back. 

## To-do

I found a bug about focus while I was working on this: https://github.com/GoogleForCreators/web-stories-wp/issues/11196

## User-facing changes

If I did this right, then there are no user facing changes. 

## Testing Instructions

Please make sure that the publish flow is still working as expected for admins and contributors.
 As well as:  publishing with different visibility and going from story details modal to review checklist.




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #10470
